### PR TITLE
Check input length when unmarshaling non-WS curves

### PR DESCRIPTION
**Pull request by ghedo**

### CODE REVIEW

File: src/crypto/elliptic/elliptic.go [hunk 1]
```diff
@@ -339,7 +339,10 @@ func Unmarshal(curve Curve, data []byte) (x, y *big.Int) {
 	byteLen := (curve.Params().BitSize + 7) >> 3
 
 	if _, ok := curve.(nonWSCurve); ok {
-		x = new(big.Int).SetBytes(data[0:byteLen])
+		if len(data) != byteLen {
+			return
+		}
+		x = new(big.Int).SetBytes(data[:byteLen])
```
Comment (FiloSottile) [inline src/crypto/elliptic/elliptic.go]: No need for `[:byteLen]` here.

Comment (vkrasnov) [inline src/crypto/elliptic/elliptic.go]: I think so too

Comment (ghedo) [inline src/crypto/elliptic/elliptic.go]: Ok, fixed (I think that's what you meant anyway).

**Comment by FiloSottile** [comment]
Thanks!

**Pull request merged**